crates/oxc_allocator/src/boxed.rs

```diff
@@ -76,6 +76,18 @@ impl<'alloc, T> Box<'alloc, T> {
     }
 }
 
+impl<'alloc, T: ?Sized> Box<'alloc, T> {
+    /// Create a [`Box`] from a raw pointer to a value in the arena.
+    ///
+    /// # SAFETY
+    /// It is only safe to create a `Box` this way with pointers taken out of
+    /// another `Box` in the same arena.
```

Comment:
This comment should be updated. It doesn't require that the pointer is from *another `Box`*. If that was true, then `Vec:: into_boxed_slice` which you're adding here would violate the contract, as it's not a `Box`!

The requirement is that pointer needs to point to data in same arena (could be from a `Vec`, a `String`, anything else).

Actually, not even that is quite right. We *want* it to be date from the same arena, but the safety constraint I think is  a bit looser - it can point to *any* data as long as that data outlives `'alloc`.

There's also a safety requirement about no aliasing.

This may seem like nitpicking, but it's vitally important that documentation for unsafe methods is accurate about what the contract is, otherwise it's impossible to verify the method is being used correctly.

Reply:
I've implemented this change in an extra commit.